### PR TITLE
Add artifact filters to deployment jobs

### PR DESCRIPTION
**Pull request by hallipr**
To reduce build time and disk space usage, prevent deployment jobs from automatically downloading all pipeline assets.  Only download the assets and asset contents required by each job.

### CODE REVIEW

File: eng/pipelines/templates/stages/archetype-java-release-pom-only.yml [hunk 1]
```diff
@@ -63,6 +63,10 @@ stages:
               runOnce:
                 deploy:
                   steps:
+                    - download: none
```
Comment (weshaggard) [inline eng/pipelines/templates/stages/archetype-java-release-pom-only.yml:66]: Did you verify if this is required if we explicitly list the download artifacts?

**Review by weshaggard — APPROVED** [review]

File: eng/pipelines/templates/stages/archetype-java-release-pom-only.yml (hunk 1 above)
Comment (hallipr) [reply eng/pipelines/templates/stages/archetype-java-release-pom-only.yml:66]: I'm checking functionality with -download: none first.   If it's all good, I'll try removing it.
https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1229360&view=logs&s=69878f40-1e39-53f0-a439-635d139ede3d

Comment (hallipr) [reply eng/pipelines/templates/stages/archetype-java-release-pom-only.yml]: I verified that adding an explicit download step removes the implicit step.  There's no need to add `download: none` before `download: current`

**Comment by hallipr** [comment]
Verified using template pipeline:
https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1229360&view=logs&j=7bb69cc7-6f52-585e-8b13-5bae7c4a9fa4&t=7bb69cc7-6f52-585e-8b13-5bae7c4a9fa4

**Review by danieljurek — APPROVED** [review]
:rocket:

**Pull request merged**